dotnet/src/Extensions/Reliability.Basic/KernelConfigExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+#pragma warning disable IDE0130 // Namespace does not match folder structure
+
+using Microsoft.SemanticKernel.Reliability.Basic;
+
+namespace Microsoft.SemanticKernel;
+
+/// <summary>
+/// Extension methods for <see cref="KernelConfig"/> class.
+/// </summary>
+public static class KernelConfigExtensions
+{
+    /// <summary>
+    /// Sets the default retry configuration for any kernel http request.
+    /// </summary>
+    /// <param name="kernelConfig">Target instance</param>
+    /// <param name="retryConfig">Retry configuration</param>
+    /// <returns>Returns target instance for fluent compatibility</returns>
+    public static KernelConfig SetRetryBasic(this KernelConfig kernelConfig, BasicRetryConfig? retryConfig = null)
```

Comment:
Since this is the last thing in KernelConfig, I think we should kill it completely and add extensions for KernelBuilder.WithBasicRetryHandler(BasicRetryConfig config)